### PR TITLE
Enable and upgrade `stylelint` in `pre-commit`

### PR DESCRIPTION
**Pull request by Darwinkel**
### Changes
It was planned to integrate `stylelint` as part of the Manon upgrade. The pipeline was implemented, but actually fixing the styling issues has not been done yet. This PR upgrades and enables it.

I'd like to request the frontend-gods @TwistMeister or @HeleenSG to perform the actual fixes :)

Running `pre-commit run --all-files` should already do most of the work.

### Issue link
Originally mentioned in https://github.com/minvws/nl-kat-coordination/pull/548 and https://github.com/minvws/nl-kat-coordination/pull/719.

### Proof
See CI.

---

### Code Checklist
- [ ] All the commits in this PR are properly PGP-signed and verified;
- [ ] This PR only contains functionality relevant to the issue; tickets have been created for newly discovered issues.
- [ ] I have written unit tests for the changes or fixes I made.
- [ ] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [ ] I have performed a self-review of my code and refactored it to the best of my abilities.

### Communication
- [ ] I have informed others of any required `.env` changes files if required and changed the `.env-dist` accordingly.
- [ ] I have made corresponding changes to the documentation, if necessary.

---
## Checklist for code reviewers:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---
## Checklist for QA:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

**Review by ammar92 — DISMISSED** [review]

File: yarn.lock
[the remark this answers is not in the document]
Comment (TwistMeister) [reply yarn.lock:1]: Deleted the accidentally added and empty `yarn.lock` file.

File: boefjes/tests/modules/dummy_boefje_environment_with_pycache/__pycache__/pytest__init__.cpython-311.pyc
[the remark this answers is not in the document]
Comment (Darwinkel) [reply boefjes/tests/modules/dummy_boefje_environment_with_pycache/__pycache__/pytest__init__.cpython-311.pyc:1]: Yes. Not sure how this happened.

**Review by praseodym — APPROVED** [review]

**Review by praseodym — CHANGES_REQUESTED** [review]
## Checklist for QA:
- [x] I have checked out this branch, and successfully ran a fresh `make reset`.
- [x] I confirmed that there are no unintended functional regressions in this branch:
    - [x] I have managed to pass the onboarding flow
    - [x] Objects and Findings are created properly
    - [x] Tasks are created and completed properly
- [ ] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.

### What works:
General functionality excluding styling.

### What doesn't work:
Header is broken:
Screenshot 2023-07-19 at 09-30-20 OpenKAT - landing_page
Screenshot 2023-07-19 at 09-31-41 OpenKAT - step_setup_scan_ooi_detail
Crisis room layout is weird:
Screenshot 2023-07-19 at 09-34-56 OpenKAT - crisis_room
And other styling issues.

### Bug or feature?:
* _bullet point + screenshot (if useful) if it is unclear whether something is a bug or an intended feature._

**Comment by TwistMeister** [comment]
@praseodym can you verify if this persists when you run `yarn install` and `yarn build` and after that "clear cache and hard reload" the page, with your dev tools open so that we're sure all the correct styles are loaded?

**Comment by Darwinkel** [comment]
> @praseodym can you verify if this persists when you run `yarn install` and `yarn build` and after that "clear cache and hard reload" the page, with your dev tools open so that we're sure all the correct styles are loaded?
> 
> Edit: When I do this, I see the correct styling for the header. The duplicate submit buttons isn't correct but also isn't regression, since it's been that way for a while now, so I would like to address that in another ticket.

@praseodym and I just verified that the discrepancy is caused by old `dist`/`node_modules` files. Removing those folders and running `make reset` reproduces the below finding:

image

**Comment by praseodym** [comment]
Just checked out a clean copy of this branch, the header is still broken.

**Review by praseodym — CHANGES_REQUESTED** [review]
Just checked out a clean copy of this branch, the header is still broken.

**Comment by praseodym** [comment]
@HeleenSG JavaScript is indeed still broken, I see the error
`[Error] Failed to load resource: the server responded with a status of 404 () (manon.js, line 0)` in my browser log and the login page looks like this:
Screenshot 2023-08-07 at 11 22 26

**Comment by ammar92** [comment]
I have just checked out this branch and can't reproduce any of the errors mentioned here before. I've cleared the Parcel cache and can compile and load `manon.js` from Rocky. Help text and the main menu works as well.

**Comment by praseodym** [comment]
You probably have some old assets still left over in your Django static directory.

After building 4c481e26 in a completely clean environment, `/static/dist/manon.js` can't be found and the login page looks like this:
Screenshot 2023-08-29 at 14 35 58

**Comment by ammar92** [comment]
> You probably have some old assets still left over in your Django static directory.
> 
> After building [4c481e2](https://github.com/minvws/nl-kat-coordination/commit/4c481e2663d68e7c13053b47ee07e280cbbed0f0) in a completely clean environment, `/static/dist/manon.js` can't be found and the login page looks like this: Screenshot 2023-08-29 at 14 35 58

You're right, I managed to reproduce this with a freshly checked-out repository. After some debugging and discussion with @HeleenSG we came to the conclusion that `manon.js` wasn't properly linked in `rocky/templates/head.html` after some restructuring of the `assets` directory. It should be fixed now.

**Review by underdarknl — APPROVED** [review]

**Review by underdarknl — APPROVED** [review]

**Review by sigio — APPROVED** [review]

**Pull request merged**